device-bridge/devices/nodemcu-minion.js

```diff
@@ -72,16 +75,28 @@ NodemcuMinion.prototype.off = function() {
 	} ) );
 };
 
+NodemcuMinion.prototype.onColorChange = function( color ) {
+	var topic = 'iot/things/' + this.id.split( '/' )[ 1 ]; //Remove gateway reference and substitute with iot/things
+	this.client.publish( topic, JSON.stringify( {
+		action: 'set',
+		state: color
+	} ) );
+};
+
+
 NodemcuMinion.prototype.processQueueTask = function( data, progress, resolve, reject ) {
-	//to replace later
+	//to replace later   
 	if ( data.action === 'set' ) {
 		this.forwardToDevice( data );
 		resolve();
-	} else 	if ( data.action === 'off' ) {
+	} else if ( data.action === 'off' ) {
 		this.off();
 		resolve();
-	} else {
-		reject( 'Unknown command' );
+	} else if ( data.action === 'gradient' ) {
+        this.animation.gradient( this.state, data.state, data.duration, this.onColorChange.bind( this ), function() { resolve(); } );
```

Comment:
If that funcion was a promise, you could:

```
animateGradient( this.state, data.state, data.duration, newColor => forwardToDevice( { action: "set", state: newColor } )  ).then( resolve );
```

And you could delete `NodemcuMinion.prototype.onColorChange`